gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -680,6 +680,140 @@ gst_tensor_filter_common_free_property (GstTensorFilterPrivate * priv)
+    else if (g_ascii_strcasecmp (file_extension, ".bin") == 0
+        || g_ascii_strcasecmp (file_extension, ".xml") == 0)
+      detected_fw = g_strdup ("openvino");
+    else {
+      g_warning ("%s is not supported or invalid extension \n", file_extension);
+      return detected_fw;
+    }
+  } else if (num_models == 2) {
+    file_extension_2 = strrchr (model_files[1], '.');
+    if ((g_ascii_strcasecmp (file_extension, ".so") == 0
+            && g_ascii_strcasecmp (file_extension_2, ".nb") == 0)
+        || (g_ascii_strcasecmp (file_extension_2, ".so") == 0
+            && g_ascii_strcasecmp (file_extension, ".nb") == 0))
+      detected_fw = g_strdup ("vivante");
+    else {
+      g_warning ("%s is not supported or invalid extension \n", file_extension);
+      return detected_fw;
+    }
+  } else {
+    g_warning ("Invalid the number of models");
+    return detected_fw;
+  }
+
+  /* .pb or .tflite have multiple framework. */
+  if (fw_priority_str != NULL) {
+    fw_priority_str_arr = g_strsplit (fw_priority_str, ",", -1);
+
+    for (i = 0; i < g_strv_length (fw_priority_str_arr); i++) {
+      fw = nnstreamer_filter_find (fw_priority_str_arr[i]);
+
+      if (fw) {
+        detected_fw = g_strdup (fw_priority_str_arr[i]);
+        break;
+      } else {
+        g_debug ("Cannot identify the neural network framework. %s\n",
+            fw_priority_str_arr[i]);
+      }
+    }
+    if (fw == NULL) {
+      g_warning ("Cannot get any neural network framework of %s. \n",
+          file_extension);
+      detected_fw = NULL;
+    }
+
+    g_free (fw_priority_str);
+    g_strfreev (fw_priority_str_arr);
+  }
+
+  g_debug ("Automatically detected framework is %s.\n", detected_fw);
+  g_debug
+      ("If you want to change priority of framerwork for auto detection, \n");
+  g_debug ("Please edit meson_option.txt \n");
+  g_debug ("You can find the file at root path of nnstreamer. \n");
+
+  return detected_fw;
+}
+
+/**
+ * @brief automatically selecting framework for tensor filter
+ * @param[in] priv Struct containing the properties of the object
+ * @param[in] fw_name Framework name
+ */
+void
+gst_tensor_filter_get_available_framework (GstTensorFilterPrivate * priv,
+    const char *fw_name)
+{
+  GstTensorFilterProperties *prop;
+  const GstTensorFilterFramework *fw;
+  gchar *detected_fw = NULL;
+
+  prop = &priv->prop;
+
+  /* If model file is not loaded, get framework after loading the model */
+  if (prop->model_files == NULL && g_strcmp0 (fw_name, "auto") == 0) {
+    prop->fwname = g_strdup (fw_name);
+    return;
+  }
+
+  /* Find appopriate framework of the given model when framework auto-detecion mode */
+  if (g_strcmp0 (fw_name, "auto") == 0) {
+    detected_fw =
+        gst_tensor_filter_framework_auto_detection (prop->model_files,
+        prop->num_models);
+    fw_name = detected_fw;
+  }
+
+  fw = nnstreamer_filter_find (fw_name);
+
+  if (fw) {
+    priv->fw = fw;
+    prop->fwname = g_strdup (fw_name);
```

Comment:
possible mem leak from line 791. add g_free() and init null in line 804.

Reply:
Thank you. Updated!